			     BASH PATCH REPORT
			     =================

Bash-Release: 4.0
Patch-ID: bash40-021

Bug-Reported-by:  Matt Zyzik <matt.zyzik@nyu.edu>
Bug-Reference-ID: <20090319015542.696F62B8E8@ice.filescope.com>
Bug-Reference-URL: http://lists.gnu.org/archive/html/bug-bash/2009-03/msg00149.html

Bug-Description:

When not in a locale supporting multibyte characters, readline will occasionally
not erase characters between the cursor position and the end of the line
when killing text backwards.

Patch:

*** bash-4.0/lib/readline/display.c	2009-01-04 14:32:32.000000000 -0500
--- bash-4.0-patched/lib/readline/display.c	2009-04-14 14:00:18.000000000 -0400
***************
*** 1775,1779 ****
  	     adjust col_lendiff based on the difference between _rl_last_c_pos
  	     and _rl_screenwidth */
! 	  if (col_lendiff && (_rl_last_c_pos < _rl_screenwidth))
  #endif
  	    {	  
--- 1775,1779 ----
  	     adjust col_lendiff based on the difference between _rl_last_c_pos
  	     and _rl_screenwidth */
! 	  if (col_lendiff && ((MB_CUR_MAX == 1 || rl_byte_oriented) || (_rl_last_c_pos < _rl_screenwidth)))
  #endif
  	    {	  
*** bash-4.0/patchlevel.h	2009-01-04 14:32:40.000000000 -0500
--- bash-4.0-patched/patchlevel.h	2009-02-22 16:11:31.000000000 -0500
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 20
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 21
  
  #endif /* _PATCHLEVEL_H_ */
